src/Spryker/Glue/AppMerchantBackendApi/Mapper/GlueRequestMerchantAppOnboardingMapper.php

```diff
@@ -18,9 +18,9 @@ public function mapGlueRequestTransferToMerchantAppOnboardingRequestTransfer(
     ): MerchantAppOnboardingRequestTransfer {
         $merchantTransfer = new MerchantTransfer();
         $merchantTransfer->fromArray($glueRequestTransfer->getAttributes(), true);
+        $merchantTransfer->fromArray($glueRequestTransfer->getAttributes()['merchant'] ?? [], true);
         $merchantTransfer->setTenantIdentifier($glueRequestTransfer->getMeta()['x-tenant-identifier'][0] ?? '');
         $merchantTransfer->setMerchantReference($glueRequestTransfer->getMeta()['x-merchant-reference'][0] ?? '');
```

Comment:
it will make `merchantReference` empty when is not sent via header.

```suggestion
        if (empty($glueRequestTransfer->getMeta()['x-merchant-reference'][0])) {
            $merchantTransfer->setMerchantReference($glueRequestTransfer->getMeta()['x-merchant-reference'][0]);
       }
```

Reply:
or it's still sent by MerchantApp module?

Reply:
it's still send. but I'll add it